### PR TITLE
fix(dispatcher): only download materials that are in the CAS

### PR DESCRIPTION
**Pull request by migmartri**
This fixes an issue where the dispatcher fails if there are attached integrations in a workflow and one of the associated materials was a container image type.

This change makes sure that the dispatcher only downloads the materials that are in fact in the CAS. Before it used to infer a downloadable item if there was a `digest` and no content available. This is wrong, since container images match this criteria and are not in the CAS. 

To solve it I added an additional annotation to the materials that is set once its uploaded to the CAS. To do that I leverage the common download and craft method introduced in #120

image

I also made sure that all the uploadable materials are now using such method and hence the new annotation. 

Refs #38

### CODE REVIEW

File: app/controlplane/internal/dispatcher/dispatcher.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ func (d *FanOutDispatcher) Run(ctx context.Context, envelope *dsse.Envelope, org
 		// Retrieve material content
 		content := []byte(material.Value)
 		// It's a downloadable so we retrieve and override the content variable
-		if material.Hash != nil {
+		if material.Hash != nil && material.DownloadableFromCAS {
```
Comment (danlishka) [inline app/controlplane/internal/dispatcher/dispatcher.go:202]: Is this check needed here anymore?

```
material.Hash != nil
```

File: internal/attestation/crafter/materials/junit_xml_test.go [hunk 2]
```diff
@@ -45,7 +46,7 @@ func TestNewJUnitXMLCrafter(t *testing.T) {
 		{
 			name: "wrong type",
 			input: &contractAPI.CraftingSchema_Material{
-				Type: contractAPI.CraftingSchema_Material_CONTAINER_IMAGE,
+				Type: contractAPI.CraftingSchema_Material_SBOM_CYCLONEDX_JSON,
```
Comment (danlishka) [inline internal/attestation/crafter/materials/junit_xml_test.go:49]: Why did we change it here?

File: internal/attestation/renderer/chainloop/v02.go [hunk 3]
```diff
@@ -194,6 +199,10 @@ func normalizeMaterial(material *slsa_v1.ResourceDescriptor) (*NormalizedMateria
 		return nil, fmt.Errorf("material name not found")
 	}
 
+	if v, ok := material.Annotations[AnnotationMaterialCAS]; ok && v.(bool) {
+		m.DownloadableFromCAS = true
```
Comment (migmartri) [inline internal/attestation/renderer/chainloop/v02.go:203]: change to uploaded to CAS

Comment (danlishka) [reply internal/attestation/renderer/chainloop/v02.go:203]: Based on our conversation in the morning. Yes, it will be more consistent this way.

**Review by danlishka — APPROVED** [review]
LGTM

**Pull request merged**